tests.ml

```diff
@@ -1,28 +1,80 @@
+  Alcotest.(check bool "longest label is domain name" true (is_domain longest_label)) ;
+  Alcotest.(check bool "longest label + a is not domain name" false (is_domain (longest_label ^ "a"))) ;
+  Alcotest.(check bool "ll.ll.ll.ll[:-2] is domain name" true
+              (is_domain (longest_prefix ^ "." ^ (String.sub longest_label 0 61)))) ;
+  Alcotest.(check bool "ll.ll.ll.ll[:-1] is not a domain name" false
+              (is_domain (longest_prefix ^ "." ^ (String.sub longest_label 0 62)))) ;
+  Alcotest.(check bool "foo._bar is not a hostname" false (is_host (n_of_s "foo._bar"))) ;
+  Alcotest.(check bool "2foo.bar is a hostname" true (is_host (n_of_s "2foo.bar"))) ;
+  Alcotest.(check bool "f2.bar is a hostname" true (is_host (n_of_s "f2.bar"))) ;
+  Alcotest.(check bool "-f2.bar is not a hostname" false (is_host (n_of_s "-f2.bar"))) ;
+  Alcotest.(check bool "f2.23 is not a hostname" false (is_host (n_of_s "f2.23"))) ;
+  Alcotest.(check bool "42.23b is a hostname" true (is_host (n_of_s "42.23b"))) ;
+  Alcotest.(check bool "'bar.foo is not a hostname" false (is_host (n_of_s "'bar.foo"))) ;
+  Alcotest.(check bool "root is no service" false (is_service Domain_name.root)) ;
   Alcotest.(check bool "_tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "_tcp.foo"))) ;
+              (is_service (n_of_s "_tcp.foo"))) ;
   Alcotest.(check bool "_._tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "_._tcp.foo"))) ;
+              (is_service (n_of_s "_._tcp.foo"))) ;
   Alcotest.(check bool "foo._tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "foo._tcp.foo"))) ;
+              (is_service (n_of_s "foo._tcp.foo"))) ;
   Alcotest.(check bool "f_oo._tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "f_oo._tcp.foo"))) ;
+              (is_service (n_of_s "f_oo._tcp.foo"))) ;
   Alcotest.(check bool "foo_._tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "foo_._tcp.foo"))) ;
+              (is_service (n_of_s "foo_._tcp.foo"))) ;
   Alcotest.(check bool "_xmpp-server._tcp.foo is a service" true
-              (Domain_name.is_service (n_of_s ~hostname:false "_xmpp-server._tcp.foo"))) ;
+              (is_service (n_of_s "_xmpp-server._tcp.foo"))) ;
   Alcotest.(check bool "_xmpp-server._tcp2.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "_xmpp-server._tcp2.foo"))) ;
+              (is_service (n_of_s "_xmpp-server._tcp2.foo"))) ;
   Alcotest.(check bool "_xmpp_server._tcp.foo is no service" false
-              (Domain_name.is_service (n_of_s ~hostname:false "_xmpp_server._tcp.foo"))) ;
+              (is_service (n_of_s "_xmpp_server._tcp.foo"))) ;
   Alcotest.(check bool "_xmpp-server-server._tcp.foo is no service" false
```

Comment:
Unrelated to code change, but the comment `_xmpp-server` is different from the used actually used value `_xmpp_server` (underscore/dash)

Reply:
thanks, fixed